microbench/src/main/java/io/netty/microbench/http2/NoPriorityByteDistributionBenchmark.java

```diff
@@ -0,0 +1,315 @@
+        restoreStreamWindows();
+    }
+
+    private void restoreStreamWindows() throws Http2Exception {
+        restoreWindowVisitor.visit(connection.connectionStream());
+        connection.forEachActiveStream(restoreWindowVisitor);
+    }
+
+    private void addData(Http2Stream stream, final int dataSize) {
+        controller.addFlowControlled(stream, new Http2RemoteFlowController.FlowControlled() {
+            int size = dataSize;
+
+            @Override
+            public int size() {
+                return size;
+            }
+
+            @Override
+            public void error(ChannelHandlerContext ctx, Throwable cause) {
+                cause.printStackTrace();
+            }
+
+            @Override
+            public void writeComplete() {
+                // Don't care.
+            }
+
+            @Override
+            public void write(ChannelHandlerContext ctx, int allowedBytes) {
+                size -= allowedBytes;
+            }
+
+            @Override
+            public boolean merge(ChannelHandlerContext ctx,
+                                 Http2RemoteFlowController.FlowControlled next) {
+                // Don't merge.
+                return false;
+            }
+        });
+    }
+
+    private static int toStreamId(int i) {
+        return 2 * i + 1;
+    }
+
+    private class ByteCounter implements StreamByteDistributor {
+        private final StreamByteDistributor delegate;
+        private int totalBytesWritten;
+        private int minBytesWritten;
+        private int maxBytesWritten;
+
+        private ByteCounter(StreamByteDistributor delegate) {
+            this.delegate = delegate;
+            clear();
+        }
+
+        void clear() {
+            totalBytesWritten = 0;
+            minBytesWritten = Integer.MAX_VALUE;
+            maxBytesWritten = 0;
+        }
+
+        @Override
+        public void updateStreamableBytes(StreamState state) {
+            delegate.updateStreamableBytes(state);
+        }
+
+        @Override
+        public boolean distribute(int maxBytes, Writer writer)
+                throws Http2Exception {
+            return delegate.distribute(maxBytes, new CountingWriter(writer));
+        }
+
+        private class CountingWriter implements Writer {
+            private final Writer delegate;
+
+            private CountingWriter(Writer delegate) {
+                this.delegate = delegate;
+            }
+
+            @Override
+            public void write(Http2Stream stream, int numBytes) {
+                totalBytesWritten += numBytes;
+                if (numBytes > 0 && numBytes < minBytesWritten) {
+                    minBytesWritten = numBytes;
+                }
+                if (numBytes > maxBytesWritten) {
+                    maxBytesWritten = numBytes;
+                }
+
+                delegate.write(stream, numBytes);
+            }
+        }
+    }
+
+    private static class TestChannel extends AbstractChannel {
```

Comment:
Mocking channel ends up being a pain with all of the current logic in `DefaultHttp2RemoteFlowController`. It uses a lot of the interface and we've moved away from mocking it in http2 for that reason.